parser/prism/Translator.cc

```diff
@@ -514,7 +514,15 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
                         receiverExpr = receiver->takeDesugaredExpr();
                     }
 
-                    flags.isPrivateOk = PM_NODE_FLAG_P(callNode, PM_CALL_NODE_FLAGS_IGNORE_VISIBILITY);
+                    // Unsupported nodes are desugared to an empty tree.
```

Comment:
There were tests that cover this behaviour, but they weren't hitting the direct desugar code-path because they were blocked by support for statements nodes.

Now that those are handled, this issue was uncovered, which is why this change is in this PR.

Reply:
Discussed further [here](https://github.com/sorbet/sorbet/pull/9274#discussion_r2325114575).